node/network/bridge/src/validator_discovery.rs

```diff
@@ -115,39 +114,93 @@ impl NonRevokedConnectionRequestState {
 	}
 }
 
+/// Will be called by [`Service::on_request`] when a request was revoked.
+///
+/// Takes the `map` of requested validators and the `id` of the validator that should be revoked.
+///
+/// Returns `Some(id)` iff the request counter is `0`.
+fn on_revoke(map: &mut HashMap<AuthorityDiscoveryId, u64>, id: AuthorityDiscoveryId) -> Option<AuthorityDiscoveryId> {
+	if let hash_map::Entry::Occupied(mut entry) = map.entry(id) {
+		if entry.get_mut().saturating_sub(1) == 0 {
+			return Some(entry.remove_entry().0);
+		}
+	}
+
+	None
+}
+
+fn peer_id_from_multiaddr(addr: &Multiaddr) -> Option<PeerId> {
```

Comment:
is this trying to do what https://github.com/paritytech/substrate/blob/30ecb9b259140a860b20c662fda01b5b2c33e5fa/client/network/src/service.rs#L1067-L1085 does?

I guess the only difference is `local_peer_id` handling, I don't know if it could be an issue here.

Reply:
This function is directly copied from authority discovery ;)

Reply:
hopefully we don't receive local peer id when peer is connected :)